Server/Server.Core/Server.Core.Static/Server.cpp

```diff
@@ -145,131 +259,50 @@ void Server::onMessage(const std::shared_ptr<Connection>& client, Message& messa
         }
     }
 }
-Server::Server(const uint16_t& port) :
-     _acceptor(_context, tcp::endpoint(tcp::v4(), port)),
-     _postgreManager(std::make_unique<PostgreRepositoryManager>(PostgreAdapter::getInstance<PostgreAdapter>()))
-{
-}
 
-Server::~Server() { stop(); }
-
-void Server::start()
+void Server::messageClient(std::shared_ptr<Connection> client, const Message& message) const
 {
-    waitForClientConnection();
-
-    size_t threadsCount = std::thread::hardware_concurrency();
-    threadsCount > 1 ? --threadsCount : threadsCount = 1;
-    
-    for (size_t i = 0; i < threadsCount; ++i)
+    if (client != nullptr && client->isConnected())
     {
-        _threads.emplace_back(std::thread([this]() { _context.run(); }));
+        client->send(message);
     }
-    _postgreManager->handleRequests();
-
-    FileLogger::getInstance().log
-    (
-        "[SERVER] Started!", 
-        LogLevel::INFO
-    );
-}
-
-void Server::stop()
-{
-    _context.stop();
-
-    _postgreManager->stopHandler();
-
-    for (std::thread& thread : _threads)
+    else
     {
-        if (thread.joinable())
-        {
-            thread.join();
-        }
-    }
+        onClientDisconnect(client);
 
-    _threads.clear();
+       // client.reset();
```

Comment:
I think the thing to do here is to send a message about the wrong message, and so collect logs without putting the wrong message in the container.

Reply:
The server could be constructed using an already existing object. That's the idea of builder pattern. Other question why I use here qualified type instead of interface type